src/main/java/teammates/ui/controller/InstructorCommentsPageData.java

```diff
@@ -126,4 +195,95 @@ public boolean isInstructorAllowedForPrivilegeOnComment(CommentAttributes commen
+            
+            String editedAt = comment.getEditedAtTextForInstructor(giverEmailToGiverNameMap.get(giverEmail).equals("Anonymous"));
+            
+            String showCommentsTo = getShowCommentsToForComment(comment);
+            
+            String showGiverNameTo = getShowGiverNameToForComment(comment);
+            
+            String showRecipientNameTo = getShowRecipientNameToForComment(comment);
+            
+            VisibilityCheckboxes visibilityCheckboxes = createVisibilityCheckboxes(comment);
+            
+            rows.add(new InstructorCommentsCommentRow(giverEmail, comment, recipientDetails, creationTime, 
+                                 isInstructorAllowedToModifyCommentInSection, typeOfPeopleCanViewComment, editedAt,
+                                 visibilityCheckboxes, showCommentsTo, showGiverNameTo, showRecipientNameTo));
+        }       
+        return rows;
+    }
+    
+    private VisibilityCheckboxes createVisibilityCheckboxes(CommentAttributes comment) {
+        boolean isRecipientAbleToSeeComment = comment.showCommentTo.contains(CommentParticipantType.PERSON);
+        boolean isRecipientAbleToSeeGiverName = comment.showGiverNameTo.contains(CommentParticipantType.PERSON);
+        
+        boolean isRecipientTeamAbleToSeeComment = comment.showCommentTo.contains(CommentParticipantType.TEAM);
+        boolean isRecipientTeamAbleToSeeGiverName = comment.showGiverNameTo.contains(CommentParticipantType.TEAM);
+        boolean isRecipientTeamAbleToSeeRecipientName = comment.showRecipientNameTo.contains(CommentParticipantType.TEAM);
+        
+        boolean isRecipientSectionAbleToSeeComment = comment.showCommentTo.contains(CommentParticipantType.SECTION);
+        boolean isRecipientSectionAbleToSeeGiverName = comment.showGiverNameTo.contains(CommentParticipantType.SECTION);
+        boolean isRecipientSectionAbleToSeeRecipientName = comment.showRecipientNameTo.contains(CommentParticipantType.SECTION);
+        
+        boolean isCourseStudentsAbleToSeeComment = comment.showCommentTo.contains(CommentParticipantType.COURSE);
+        boolean isCourseStudentsAbleToSeeGiverName = comment.showGiverNameTo.contains(CommentParticipantType.COURSE);
+        boolean isCourseStudentsAbleToSeeRecipientName = comment.showRecipientNameTo.contains(CommentParticipantType.COURSE);
+        
+        boolean isInstructorsAbleToSeeComment = comment.showCommentTo.contains(CommentParticipantType.INSTRUCTOR);
+        boolean isInstructorsAbleToSeeGiverName = comment.showGiverNameTo.contains(CommentParticipantType.INSTRUCTOR);
+        boolean isInstructorsAbleToSeeRecipientName = comment.showRecipientNameTo.contains(CommentParticipantType.INSTRUCTOR);
+        
+        return new VisibilityCheckboxes(isRecipientAbleToSeeComment, isRecipientAbleToSeeGiverName,
+                isRecipientTeamAbleToSeeComment, isRecipientTeamAbleToSeeGiverName, isRecipientTeamAbleToSeeRecipientName,
```

Comment:
Considering these are all reliable on `comment`, why not just pass `comment` in and deal with the boolean setting over in the other class?